tests/test_client.py

```diff
@@ -58,16 +58,40 @@ def client():
 
 
 class TestClientSecretBasic(object):
-    def test_construct(self, client):
-        cis = AccessTokenRequest(code="foo", redirect_uri="http://example.com")
+    @pytest.mark.parametrize(
+        "url_encoded, encoding",
+        ((False, "latin1"), (True, "latin1"), (None, None), (False, None)),
+    )
+    def test_construct(self, client, url_encoded, encoding):
 
+        cis = AccessTokenRequest(code="foo", redirect_uri="http://example.com")
         csb = ClientSecretBasic(client)
-        http_args = csb.construct(cis)
-        cred = "{}:{}".format(quote_plus("A"), quote_plus("boarding pass"))
+
+        user, passwd = "A", "boarding pass"
+        kwargs = {}
+        if url_encoded is None or url_encoded:
+            user, passwd = (
+                quote_plus(user),
+                quote_plus(passwd),
+            )
+
+        if url_encoded is not None:
+            kwargs.update({"url_encoded": url_encoded})
+
+        if encoding is None:
+            encoding = "utf-8"
+        else:
+            kwargs.update({"encoding": encoding})
+
+        http_args = csb.construct(cis, **kwargs)
+
+        user, passwd = user.encode(encoding), passwd.encode(encoding)
```

Comment:
This assigns `bytes` to what was preciously `str`, mypy doesn't like that it may cause programming errors.

It should be solved by using a new variable, or encoding directly in L90.

Reply:
That should actually solve both mypy complains...